internal/services/loganalytics/log_analytics_workspace_resource.go

```diff
@@ -227,6 +236,14 @@ func resourceLogAnalyticsWorkspaceCreateUpdate(d *pluginsdk.ResourceData, meta i
 		sku.Name = workspaces.WorkspaceSkuNameEnumPerGBTwoZeroOneEight
 	}
 
+	// convert features to interface{} to use 2020-08-01 version API, the 2022-10-01 version API will limit `reservation_capacity_in_gb_per_day` to 100,200,300,400,500,1000,2000,5000, which will cause a breaking change .
```

Comment:
How exactly is this an issue? In the schema `reservation_capacity_in_gb_per_day` has validation for int values from 100-5000, divisible by 100 which is the same thing here?

If someone has picked a number that is not on this allow list, the Azure API will charge them for the next tier up anyway, so they should be updating their config.

Reply:
is it acceptable that the customer has picked like 3200, and did not changed the config, then it began to fail because the validation is more strict? I thought it should be a breaking change.

Reply:
If the new version of the API only accepts these values, then we should map that in Terraform as well, even if it's a breaking change as the service team has shipped this. So the validation here https://github.com/hashicorp/terraform-provider-azurerm/blob/4190878e715d16ce8ec7564009520beafaf0e394/internal/services/loganalytics/log_analytics_workspace_resource.go#L111
Should take a list of fixed integers.

We should also then update the API version for this to the latest one since we don't gain anything from using the older version, and then we won't need to do the casting to an interface down below.